Blink/SceneDelegate.swift

```diff
@@ -91,6 +91,155 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     }
   }
   
+  /**
+   Handles the `ssh://` URL schemes and x-callback-url for devices that are running iOS 13 or higher.
+   */
+  func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
+    
+    // Handle ssh:// URL scheme
+    if let sshUrlScheme = URLContexts.first(where: { $0.url.absoluteString.starts(with: "ssh://")})?.url {
+      dump(sshUrlScheme)
+      
+      var sshCommand = "ssh"
+      
+      // Progressively unwrap all of the parameters available on the URL to form
+      // the SSH command to be later passed to the shell
+      if let port = sshUrlScheme.port {
+        sshCommand += " -p \(port)"
+      }
+      
+      if let username = sshUrlScheme.user {
+        sshCommand += " \(username)@"
+      }
+      
+      if let host = sshUrlScheme.host {
+        sshCommand += "\(host)"
+      }
+      
+      let spCtrl = _spCtrl
+      
+      guard let term = spCtrl.currentTerm() else {
+        return
+      }
+      
+      spCtrl.focusOnShellAction()
+      
+      if term.isRunningCmd() {
+        // If a SSH/mosh connection is already open in the current terminal shell
+        // create a new one and then write the SSH command
+        
+        spCtrl.newShellAction()
+        
+        guard let term = spCtrl.currentTerm() else {
+          return
+        }
+        
+        DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + 0.5) {
+          term.termDevice.write(sshCommand)
+        }
+      } else {
+        // No running command or shell found running, run the SSH command on the
+        // available shell
+        term.termDevice.write(sshCommand)
+      }
+    } else if let xCallbackUrl = URLContexts.first(where: { $0.url.absoluteString.starts(with: "blinkshell://")})?.url {
+      // Handle the x-callback-urls, format should be:
+      // blinkshell://run?key=KEY&cmd=CMD%20ENCODED
+      if let xCallbackUrlHost = xCallbackUrl.host, xCallbackUrlHost == "run" {
+        
+        let components = URLComponents(url: xCallbackUrl, resolvingAgainstBaseURL: true)
+        
+        var xCancelURL: URL?
+        var xSuccessURL: URL?
+        var xErrorURL: URL?
+        
+        guard let items = components?.queryItems else {
+          return
+        }
+        
+        if let xCancel = items.filter({ $0.name == "x-cancel" }).first?.value {
```

Comment:
it is better to use `first` method here.

`items.first({ $0.name == "x-cancel" })`